creator-node/src/snapbackSM/computeSyncModeForUserAndReplica.js

```diff
@@ -0,0 +1,117 @@
+const DBManager = require('../dbManager.js')
+const retry = require('async-retry')
+
+/**
+ * Sync mode for a (primary, secondary) pair for a user
+ */
+const SyncMode = Object.freeze({
+  None: 'NONE',
+  SecondaryShouldSync: 'SECONDARY_SHOULD_SYNC',
+  PrimaryShouldSync: 'PRIMARY_SHOULD_SYNC'
+})
+
+const FetchFilesHashNumRetries = 3
+
+/**
+ * Given user state info, determines required sync mode for user and replica. This fn is called for each (primary, secondary) pair
+ * @notice Is used when both replicas are running version >= 0.3.51
+ * @param {Object} param
+ * @param {string} param.wallet user wallet
+ * @param {number} param.primaryClock clock value on user's primary
+ * @param {number} param.secondaryClock clock value on user's secondary
+ * @param {string} param.primaryFilesHash filesHash on user's primary
+ * @param {string} param.secondaryFilesHash filesHash on user's secondary
+ * @returns {SyncMode} syncMode one of None, SecondaryShouldSync, PrimaryShouldSync
+ */
+async function computeSyncModeForUserAndReplica({
+  wallet,
+  primaryClock,
+  secondaryClock,
+  primaryFilesHash,
+  secondaryFilesHash,
+  logger
+}) {
+  if (
+    !Number.isInteger(primaryClock) ||
+    !Number.isInteger(secondaryClock) ||
+    // `null` is a valid filesHash value; `undefined` is not
+    primaryFilesHash === undefined ||
+    secondaryFilesHash === undefined
+  ) {
+    throw new Error(
+      '[computeSyncModeForUserAndReplica] Error: Missing or invalid params'
+    )
+  }
+
+  if (primaryClock === secondaryClock) {
+    if (primaryFilesHash === secondaryFilesHash) {
+      return SyncMode.None
```

Comment:
so actually I suggested early returns because then we can avoid nested if-else blocks. can we only have 1 layer of if-else checks?

Reply:
ah, fair

Reply:
addressed